engine/order_manager.go

```diff
@@ -284,6 +284,49 @@ func (m *OrderManager) validate(newOrder *order.Submit) error {
 	return nil
 }
 
+// Modify depends on the order.Modify.ID and order.Modify.Exchange fields to uniquely
+// identify an order to modify.
+func (m *OrderManager) Modify(mod *order.Modify) (*order.ModifyResponse, error) {
+	if m == nil {
+		return nil, fmt.Errorf("order manager %w", ErrNilSubsystem)
+	}
+	if atomic.LoadInt32(&m.started) == 0 {
+		return nil, fmt.Errorf("order manager %w", ErrSubSystemNotStarted)
+	}
+
+	exch := m.orderStore.exchangeManager.GetExchangeByName(mod.Exchange)
+	if exch == nil {
+		return nil, ErrExchangeNotFound
+	}
+
+	id, err := exch.ModifyOrder(mod)
+	if err != nil {
+		return nil, err
+	}
+
+	// We now have to reflect this change in GCT's internal state of managed orders.
+
+	// TODO: Current implementation is here for illustrative purposes.  It introduces
+	// race conditions.
+	det, err := m.orderStore.getByExchangeAndID(mod.Exchange, mod.ID)
+	if err != nil {
+		// TODO: How to handle that?  The order is updated on the
+		// exchange, but not found locally.
+		return nil, err
+	}
+
+	// TODO: Is using det.UpdateOrderFromModify() better in this case, since
```

Comment:
Maybe for a quick implementation we could, instead of returning an order.Detail we can have that as a field and return pointer to something like order.Managed struct which includes a mutex which will have the methods to mutate changes.